src/WLR/Types/Compositor.hsc

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE PatternSynonyms #-}
+module WLR.Types.Compositor where
+
+import Foreign.C.Types (CBool, CInt, CSize, CUInt)
+import Foreign.C.String (CString)
+import Foreign.Storable (Storable(..))
+import Foreign.Ptr (Ptr, FunPtr)
+
+import WL.Utils (WL_list)
+import WL.ServerCore (WL_resource, WL_signal, WL_listener)
+import WL.ServerProtocol (WL_output_transform)
+
+import WLR.Types.Buffer (WLR_client_buffer, WLR_buffer)
+import WLR.Render.Renderer (WLR_renderer)
+import WLR.Util.Box (WLR_fbox)
+import WLR.Util.Addon (WLR_addon_set)
+
+import PIXMAN.Pixman (PIXMAN_region32)
+
+#define WLR_USE_UNSTABLE
+#include <wlr/types/wlr_compositor.h>
+
+-- not exactly sure what int32_t should map to, but I think it's just CInt
+-- some of the fields in wlr_surface had this `int32_t` type
```

Comment:
`int32_t` corresponds to `Int32` on the Haskell side. `CInt` corresponds to C `int`, which has no fixed size.